soh/src/overlays/actors/ovl_Bg_Jya_Lift/z_bg_jya_lift.c

```diff
@@ -96,8 +96,19 @@ void BgJyaLift_DelayMove(BgJyaLift* this, GlobalContext* globalCtx) {
     if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F) || (this->moveDelay > 0)) {
         this->moveDelay++;
         if (this->moveDelay >= 20) {
-            OnePointCutscene_Init(globalCtx, 3430, -99, &this->dyna.actor, MAIN_CAM);
-            BgJyaLift_SetupMove(this);
+            // The cutscene of the platform lowering will show the central room in an unloaded state if
+            // Link is not standing on the platform as it lowers. Therefore check for the Sunlight arrows
+            // enhancement and if it's enabled, check that Link is on the platform. Otherwise skip it.
+            if (gSaveContext.n64ddFlag && CVar_GetS32("gSunlightArrows", 0)) {
+                if (GET_PLAYER(globalCtx)->actor.world.pos.x >   -19.0f && GET_PLAYER(globalCtx)->actor.world.pos.x <   139.0f &&
+                    GET_PLAYER(globalCtx)->actor.world.pos.z > -1172.0f && GET_PLAYER(globalCtx)->actor.world.pos.z < -1009.0f) {
+                    OnePointCutscene_Init(globalCtx, 3430, -99, &this->dyna.actor, MAIN_CAM);
+                }
+                BgJyaLift_SetupMove(this);
+            } else {
+                OnePointCutscene_Init(globalCtx, 3430, -99, &this->dyna.actor, MAIN_CAM);
+                BgJyaLift_SetupMove(this);
+            }
```

Comment:
couple things here, for one, it's not clear why we're checking for rando'd and sunlight arrows

also, i'm pretty sure we can reduce copypasta by restructuring the if/else stuff a bit

assuming the rando check doesn't matter, this would probably work as

having this nested this way feels a little confusing to me, i think it could work as
```c
if (!CVar_GetS32("gSunlightArrows", 0) || 
    (GET_PLAYER(globalCtx)->actor.world.pos.x > -19.0f &&
     GET_PLAYER(globalCtx)->actor.world.pos.x < 139.0f &&
     GET_PLAYER(globalCtx)->actor.world.pos.z > -1172.0f &&
     GET_PLAYER(globalCtx)->actor.world.pos.z < -1009.0f)) {
    OnePointCutscene_Init(globalCtx, 3430, -99, &this->dyna.actor, MAIN_CAM);
}
BgJyaLift_SetupMove(this);
```

Reply:
Makes sense to me!  Testing it out now

Reply:
Yep, this works and is definitely cleaner.  Done :)